src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -980,10 +993,34 @@ public INamedTypeSymbol GetTypeByMetadataName(string fullyQualifiedMetadataName)
                 }
             }
 
-            return CommonCreateTupleTypeSymbol(elementTypes, elementNames, elementLocations);
+            return CommonCreateTupleTypeSymbol(elementTypes, elementNames, elementLocations, elementNullableAnnotations);
         }
 #pragma warning restore RS0026 // Do not add multiple public overloads with optional parameters
 
+        /// <summary>
+        /// Returns a new INamedTypeSymbol with the given element types, names, and locations.
+        /// </summary>
+        public INamedTypeSymbol CreateTupleTypeSymbol(
+            ImmutableArray<ITypeSymbol> elementTypes,
+            ImmutableArray<string> elementNames,
+            ImmutableArray<Location> elementLocations)
+        {
+            return CreateTupleTypeSymbol(elementTypes, elementNames, elementLocations, elementNullableAnnotations: default);
+        }
+
+        protected static void CheckTupleElementNullableAnnotations(
+            int cardinality,
+            ImmutableArray<NullableAnnotation> elementNullableAnnotations)
+        {
+            if (!elementNullableAnnotations.IsDefault)
+            {
+                if (elementNullableAnnotations.Length != cardinality)
```

Comment:
merge ifs?

Reply:
Will leave as-is for consistency with the two `CheckTupleElement...` methods below.

---
In reply to: [301326055](https://github.com/dotnet/roslyn/pull/37048#discussion_r301326055) [](ancestors = 301326055)

Reply:
sgtm.